File: snippets/factor_call.cs
using System;
using System.Linq;
using System.Text;
using System.Numerics;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;

public class Program {
    static List<UInt64> factor (UInt64 n) {
        var pinfo = new ProcessStartInfo() {
            FileName = "factor",
            Arguments = n.ToString(),
            CreateNoWindow = true,
            UseShellExecute = false,
            RedirectStandardOutput = true
        };
        Process p = Process.Start(pinfo);
        string[] sp = p.StandardOutput.ReadToEnd().Split();
        p.WaitForExit();
        List<UInt64> r = new List<UInt64>();
        for (int i = 0; i < sp.Length - 2; ++i) { r.Add(UInt64.Parse(sp[i+1])); }
        return r;
    }
    
    static void Main (string[] args) {
        foreach (var i in factor(999999866000004473)) {
            Console.WriteLine(i);
        }
    }
}

File: snippets/factor_call.vb
Imports System
Imports System.IO
Imports System.Text
Imports System.Numerics
Imports System.Collections
Imports System.Collections.Generic
Imports System.Diagnostics

class Program
    shared function factor (ByVal n as String) as String()
        dim pinfo as new ProcessStartInfo()
        pinfo.FileName = "factor"
        pinfo.Arguments = n
        pinfo.CreateNoWindow = true
        pinfo.UseShellExecute = false
        pinfo.RedirectStandardOutput = true
        dim p as Process = Process.Start(pinfo)
        p.WaitForExit()
        dim s as String = p.StandardOutput.ReadToEnd()
        dim sp() as String = s.Split()
        dim r(sp.Length - 3) as String
        for i as Int32 = 0 to sp.Length - 3 : r(i) = sp(i+1) : next i
        return r
    end function
    
    shared sub main ()
        for each i as String in factor("999999866000004473")
            Console.WriteLine(i)
        next
    end sub
end class

